File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/Python_GPIOZero_Code/21.1.1_DHT11/DHT11.py
#!/usr/bin/env python3
#############################################################################
# Filename    : DHT11.py
# Description :	read the temperature and humidity data of DHT11
# Author      : freenove
# modification: 2024/07/29
########################################################################
import time
from Freenove_DHT import DHT      

DHTPin = 17     #define the pin of DHT11

def loop():
    dht = DHT(DHTPin)
    time.sleep(1) 
    counts = 0     # Measurement counts
    while(True):
        counts += 1
        print("Measurement counts: ", counts)
        for i in range(0,15):            
            chk = dht.readDHT11()     #read DHT11 and get a return value. Then determine whether data read is normal according to the return value.
            if (chk == 0):      #read DHT11 and get a return value. Then determine whether data read is normal according to the return value.
                print("DHT11,OK!")
                break
            time.sleep(0.1)
           
        print("Humidity : %.2f, \t Temperature : %.2f \n"%(dht.getHumidity(),dht.getTemperature()))
        time.sleep(2)   
        
if __name__ == '__main__':
    print ('Program is starting ... ')
    try:
        loop()
    except KeyboardInterrupt:
        pass
        exit()   

File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/C_Code/21.1.1_DHT11/DHT11.cpp

/**********************************************************************
* Filename    : DHT11.cpp
* Description : Read the temperature and humidity data of DHT11
* Author      : www.freenove.com
* modification: 2020/10/16
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <stdint.h>
#include "DHT.hpp"

#define DHT11_Pin  0		//define the pin of sensor

int main(){
    DHT dht;			//create a DHT class object
    int chk, counts;		//chk:read the return value of sensor; sumCnt:times of reading sensor

    printf("Program is starting ...\n");

    while (1){
        counts++; //counting number of reading times
        printf("Measurement counts : %d \n", counts);
        for (int i = 0; i < 15; i++){
            chk = dht.readDHT11(DHT11_Pin);	//read DHT11 and get a return value. Then determine whether data read is normal according to the return value.
            if(chk == DHTLIB_OK){
                printf("DHT11,OK! \n");
                break;
            }
            delay(100);
        }
        printf("Humidity is %.2f %%, \t Temperature is %.2f *C\n\n",dht.humidity, dht.temperature);
        delay(2000);
    }
    return 1;
}
